File: 300-longest-increasing-subsequence/300-longest-increasing-subsequence.java
class Solution {
    public int lengthOfLIS(int[] nums) {
        int n = nums.length;
        int[] dp = new int[n+1];
        
        Arrays.fill(dp, 1);
        dp[0] = 0;
        int res = 1;
        for(int i=1; i<=n; i++){
            for(int j=i-1; j>=1; j--){
                if(nums[i-1] > nums[j-1]){
                    dp[i] = Math.max(dp[j]+1, dp[i]);
                }
            }
            
            res = Math.max(res, dp[i]);
        }
        
        return res;
    }
}

File: 300-longest-increasing-subsequence/300-longest-increasing-subsequence.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        dp = [1 for i in range(len(nums))]
        
        max_len = 1
        for i in range(len(nums)):
            # print("nums[i] ", nums[i])
            for j in range(i-1, -1, -1):
                # print("nums[j] ", nums[j])
                if(nums[j] < nums[i]) :
                    dp[i] = max(dp[j]+1, dp[i])
            # print(i, " ", dp[i])
            max_len = max(max_len, dp[i])
            
        
        return max_len
        